sway/input/seatop_default.c

```diff
@@ -373,6 +373,9 @@ static void handle_button(struct sway_seat *seat, uint32_t time_msec,
 			seat_set_focus_layer(seat, layer);
 			transaction_commit_dirty();
 		}
+		if (state == WLR_BUTTON_PRESSED) {
+			seatop_begin_down(seat, cont, time_msec, sx, sy);
```

Comment:
Isn't `cont` guaranteed to be null here? I think we probably want a new function in `seatop_down` like `seatop_begin_down_on_layer_surface` that takes the layer surface, and a new field in `seatop_down_event`. That gives the state necessary for the `exclusive_client` check too.